### PR TITLE
Dev patch

### PR DESCRIPTION
**Pull request by Irishumanoid**
Should be merged to dev before any other PR. Baseline code.

### CODE REVIEW

**Comment by awfulbananas** [comment]
will merging this make merge conflicts with PR #31?

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java [hunk 1]
```diff
@@ -104,6 +108,10 @@ public Drivetrain() {
     SmartDashboard.putNumber("driveVelocity", 0);
   }
 
+  public ChassisSpeeds getSpeeds() {
+    return m_speeds;
```
Comment (awfulbananas) [inline src/main/java/frc/robot/subsystems/drive/Drivetrain.java:112]: where is m_speeds being updated? in drivetrain it updates in periodic, but I don't see that here, maybe just return the robot relative speeds straight from the odometry

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java [hunk 2]
```diff
@@ -234,7 +263,8 @@ private void move(Vector spdVec, double rot) {
    * @param rateLimit whether or not to use slew rate limiting
    */
   private void move(Vector spdVec, double rot, boolean rateLimit) {
-    Vector spdCommanded = spdVec;
+    spdVec = spdVec.copy();
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/drive/Drivetrain.java:266]: Hmmm why are you doing this? You are modifying a parameter here, that's usually not recommended.

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java [hunk 3]
```diff
@@ -243,7 +273,7 @@ private void move(Vector spdVec, double rot, boolean rateLimit) {
     }
 
     // Adjust input based on max speed
-    Vector spdDelivered = spdCommanded.copy().mult(DriveConfig.kMaxSpeedMetersPerSecond);
+    Vector spdDelivered = spdVec.copy().mult(DriveConfig.kMaxSpeedMetersPerSecond);
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/subsystems/drive/Drivetrain.java:276]: Why copy again? To be clear: presumably you are making a new copy with the call to `spdVec.copy()` and then making  _another_ copy with initializing `spdDelivered`. Copies are typically considered expensive operations and should usually be avoided when possible.

**Review by garrettwitowski — COMMENTED** [review]
I'm curious if we still need this PR? We have another PR open to pull in a bunch of drivetrain changes right?

File: src/main/java/frc/robot/subsystems/drive/Drivetrain.java (hunk 3 above)
Comment (Irishumanoid) [reply src/main/java/frc/robot/subsystems/drive/Drivetrain.java:276]: This is fixed in PR 32, will merge that in after this PR is tested.

Comment (awfulbananas) [reply src/main/java/frc/robot/subsystems/drive/Drivetrain.java:276]: PR #32 is about ready to be merged into drivetrain anyways, I'd rather not keep updating dev only with patch-fix branches

**Pull request closed without merging**